velox/functions/sparksql/String.h

```diff
@@ -962,15 +962,8 @@ struct ConvFunction {
   // For signed value, toBase is negative.
   static void
   toChars(out_type<Varchar>& result, int64_t signedValue, int32_t toBase) {
-    int32_t resultSize =
-        (int32_t)std::floor(
-            std::log(std::abs(signedValue)) / std::log(-toBase)) +
```

Comment:
Maybe it's better to specially handle the min value to avoid over-allocatation when resize.

> auto absValue = UNLIKELY(signedValue == std::numeric_limits<int64_t>::min()) ? std::abs(std::numeric_limits<int64_t>::max()): std::abs(signedValue)

Reply:
I do not think we can compare an overflowed int64_t with `std::numeric_limits<int64_t>::min()` or `std::numeric_limits<int64_t>::max()`.

```
(int64_t)((uint64_t)std::numeric_limits<int64_t>::max()+ 1)
=>
-9223372036854775808

(int64_t)((uint64_t)std::numeric_limits<int64_t>::min() - 1)
=>
9223372036854775807
```

Spark also uses the over-allocatation size 65, see https://github.com/apache/spark/blob/7d30bd7e25edffc84c00a057a5373c0dc4b536de/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala#L31C21-L31C38

Reply:
I see your point. But only when signedValue is `std::numeric_limits<int64_t>::min()` (no matter this value is from overflow or not), std::abs is ill-formed, not?